pkg/segment/query/processor/headcommand.go

```diff
@@ -30,11 +32,83 @@ type headProcessor struct {
 	numRecordsSent uint64
 }
 
+func (p *headProcessor) processHeadExpr(iqr *iqr.IQR) (*iqr.IQR, error) {
+	if iqr == nil || p.options.Done {
+		return nil, io.EOF
+	}
+
+	requiredFields := p.options.BoolExpr.GetFields()
+	records, err := iqr.ReadColumns(requiredFields)
+	if err != nil {
+		return nil, toputils.TeeErrorf("headProcessor.processHeadExpr: failed to read columns, requiredFields: %v, err: %v", requiredFields, err)
+	}
```

Comment:
I think if the column doesn't exist in this IQR (which is a valid case), then you'll get an error. But you'll have the check head's `null` option for what to do with these records then. We might want to change `ReadColumns()`—and probably `ReadColumn()` as well—distinguish between an `error` because the field doesn't exist and an actual error of something unexpected happening

Reply:
Rather than changing the existent methods, we can keep those as is and use the newly created method (ReadColumsnWithBackfill) in this PR wherever necessary.

Reply:
Addressed in the latest commit.